Classes/ViewHelpers/EditRecordViewHelper.php

```diff
@@ -56,10 +56,31 @@ public static function renderStatic(
                 'tools_T3monitoringT3monitor');
 
         $vars = GeneralUtility::_GPmerged('tx_t3monitoring_tools_t3monitoringt3monitor');
-        foreach ($vars as $key => $value) {
-            $parameters['returnUrl'] .= sprintf('&tx_t3monitoring_tools_t3monitoringt3monitor[%s]=%s', $key, $value);
-        }
+        $parameters['returnUrl'] .= self::buildReturnUrl($vars);
```

Comment:
You could simply use
`GeneralUtility::implodeArrayForUrl(GeneralUtility::_GPmerged('tx_t3monitoring_tools_t3monitoringt3monitor'), 'tx_t3monitoring_tools_t3monitoringt3monitor')`